pkg/propertychecker/azure/checker.go

```diff
@@ -0,0 +1,171 @@
+// requirement and checks to determine if the cluster's Azure subscription
+// and location can provision the requested VM instances.
+//
+// The cluster must have both Azure location and subscription ID labels configured.
+// Returns true if the SKU capacity requirement can be met, false otherwise.
+func (s *PropertyChecker) CheckIfMeetSKUCapacityRequirement(
+	cluster *clusterv1beta1.MemberCluster,
+	sku string,
+	capacity uint32,
+) (bool, error) {
+	location, err := labels.ExtractLabelFromMemberCluster(cluster, labels.AzureLocationLabel)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract Azure location label from cluster %s: %w", cluster.Name, err)
+	}
+
+	subID, err := labels.ExtractLabelFromMemberCluster(cluster, labels.AzureSubscriptionIDLabel)
+	if err != nil {
+		return false, fmt.Errorf("failed to extract Azure subscription ID label from cluster %s: %w", cluster.Name, err)
+	}
+
+	// Request VM size recommendations to validate SKU availability and capacity.
+	// The capacity is checked by ensuring the current allocatable capacity is greater than the requested capacity.
+	request := &computev1.GenerateAttributeBasedRecommendationsRequest{
+		SubscriptionId: subID,
+		Location:       location,
+		RegularPriorityProfile: &computev1.RegularPriorityProfile{
+			CapacityUnitType: computev1.CapacityUnitType_CAPACITY_UNIT_TYPE_VM_INSTANCE_COUNT,
+			TargetCapacity:   capacity, // CurrentAllocatableCapacity > RequestedCapacity
+		},
+		ResourceProperties: &computev1.ResourceProperties{
+			VmAttributes: &computev1.VMAttributes{
+				AllowedVmSizes: []string{sku},
+			},
+		},
+		RecommendationProperties: &computev1.RecommendationProperties{
+			RestrictionsFilter: computev1.RecommendationProperties_RESTRICTIONS_FILTER_QUOTA_AND_OFFER_RESTRICTIONS,
+		},
+	}
+
+	respObj, err := s.vmSizeRecommenderClient.GenerateAttributeBasedRecommendations(context.Background(), request)
+	if err != nil {
+		return false, fmt.Errorf("failed to generate VM size recommendations from Azure: %w", err)
+	}
+
+	// This check is a defense mechanism; vmSizeRecommenderClient should return a VM size recommendation
+	// if the SKU is available in the specified location and subscription.
+	available := false
+	for _, vm := range respObj.RecommendedVmSizes.RegularVmSizes {
+		if vm.Name == sku {
+			available = true
+			klog.V(2).Infof("SKU %s is available in cluster %s", sku, cluster.Name)
+			break
+		}
+	}
+
+	return available, nil
+}
+
+// ExtractCapacityRequirements extracts the capacity value from a PropertySelectorRequirement.
+// This function is specifically designed for Azure SKU capacity properties that follow the pattern:
+// "kubernetes.azure.com/vm-sizes/{sku}/capacity"
+// Returns the capacity as an uint32 if the requirement is valid;
+// or an error if the requirement is invalid.
+func ExtractCapacityRequirements(req placementv1beta1.PropertySelectorRequirement) (uint32, error) {
```

Comment:
```suggestion
func extractCapacityRequirements(req placementv1beta1.PropertySelectorRequirement) (uint32, error) {
```
do we need to make it public?